pkgs/by-name/af/afterglow-cursors-recolored/package.nix

```diff
@@ -0,0 +1,161 @@
+              cp -rT colors/Dracula/Orange/dist-dracula-orange $out/share/icons/Afterglow-Recolored-Dracula-Orange
+              cp -rT colors/Dracula/Pink/dist-dracula-pink $out/share/icons/Afterglow-Recolored-Dracula-Pink
+              cp -rT colors/Dracula/Purple/dist-dracula-purple $out/share/icons/Afterglow-Recolored-Dracula-Purple
+              cp -rT colors/Dracula/Red/dist-dracula-red $out/share/icons/Afterglow-Recolored-Dracula-Red
+              cp -rT colors/Dracula/Teddy/dist-dracula-teddy $out/share/icons/Afterglow-Recolored-Dracula-Teddy
+              cp -rT colors/Dracula/Yellow/dist-dracula-yellow $out/share/icons/Afterglow-Recolored-Dracula-Yellow
+
+              mkdir -p $out/share/icons/Afterglow-Recolored-Gruvbox-{Aqua,Black,Blue,Gray,Green,Mojas84,Orange,Purple,White}
+
+              cp -rT colors/Gruvbox/Aqua/dist-gruvbox-aqua $out/share/icons/Afterglow-Recolored-Gruvbox-Aqua
+              cp -rT colors/Gruvbox/Black/dist-gruvbox-black $out/share/icons/Afterglow-Recolored-Gruvbox-Black
+              cp -rT colors/Gruvbox/Blue/dist-gruvbox-blue $out/share/icons/Afterglow-Recolored-Gruvbox-Blue
+              cp -rT colors/Gruvbox/Gray/dist-gruvbox-gray $out/share/icons/Afterglow-Recolored-Gruvbox-Gray
+              cp -rT colors/Gruvbox/Green/dist-gruvbox-green $out/share/icons/Afterglow-Recolored-Gruvbox-Green
+              cp -rT colors/Gruvbox/Mojas84/dist-gruvbox-mojas84 $out/share/icons/Afterglow-Recolored-Gruvbox-Mojas84
+              cp -rT colors/Gruvbox/Orange/dist-gruvbox-orange $out/share/icons/Afterglow-Recolored-Gruvbox-Orange
+              cp -rT colors/Gruvbox/Purple/dist-gruvbox-purple $out/share/icons/Afterglow-Recolored-Gruvbox-Purple
+              cp -rT colors/Gruvbox/White/dist-gruvbox-white $out/share/icons/Afterglow-Recolored-Gruvbox-White
+
+              mkdir -p $out/share/icons/Afterglow-Recolored-Original-{Blue,Purple,joris,joris2,joris3,joris4}
+
+              cp -rT colors/Original/Blue/dist-blue $out/share/icons/Afterglow-Recolored-Original-Blue
+              cp -rT colors/Original/Purple/dist-purple $out/share/icons/Afterglow-Recolored-Original-Purple
+              cp -rT colors/Original/joris/dist-joris $out/share/icons/Afterglow-Recolored-Original-joris
+              cp -rT colors/Original/joris2/dist-joris2 $out/share/icons/Afterglow-Recolored-Original-joris2
+              cp -rT colors/Original/joris3/dist-joris3 $out/share/icons/Afterglow-Recolored-Original-joris3
+              cp -rT colors/Original/joris4/dist-joris4 $out/share/icons/Afterglow-Recolored-Original-joris4
+            ''}
+          ''}
+       ''}
+    ''}
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Recolors of the original Afterglow Cursors";
+    homepage = "https://github.com/TeddyBearKilla/Afterglow-Cursors-Recolored";
+    maintainers = with maintainers; [ d3vil0p3r ];
+    platforms = platforms.all;
+    license = licenses.gpl3Plus;
+  };
+}
```

Comment:
I think this would be saying the same with a bit less code.

That being said, I'm not convinced this level of granularity is necessary. I can't imagine anyone picking which colors the would like and not for every single kind of theme, for this one single package. It's easier to just install them all, and choose the one you're looking for in whichever theme managing solution you're using.

Unless you have a reason to need this granularity, I'd recommend prioritizing to reduce the code to a minimum, making it easier to maintain. You're the maintainer though, so I think it's up to you.

```suggestion
{ lib
, stdenvNoCC
, fetchFromGitHub
, themeVariants ? []
, catppuccinColorVariants ? []
, draculaColorVariants ? []
, gruvboxColorVariants ? []
, originalColorVariants ? []
}:

let
  pname = "afterglow-cursors-recolored";

  availableThemeVariants = [
    "Catppuccin"
    "Dracula"
    "Gruvbox"
    "Original"
  ];

  availableColorVariants = {
    Catppuccin = [
      "Blue"
      "Flamingo"
      "Green"
      "Macchiato"
      "Maroon"
      "Mauve"
      "Peach"
      "Pink"
      "Red"
      "Rosewater"
      "Sapphire"
      "Sky"
      "Teal"
      "Yellow"
    ];
    Dracula = [
      "Cyan"
      "Green"
      "Orange"
      "Pink"
      "Purple"
      "Red"
      "Teddy"
      "Yellow"
    ];
    Gruvbox = [
      "Aqua"
      "Black"
      "Blue"
      "Gray"
      "Green"
      "Mojas84"
      "Orange"
      "Purple"
      "White"
    ];
    Original = [
      "Blue"
      "Purple"
      "joris"
      "joris2"
      "joris3"
      "joris4"
    ];
  };
in

lib.checkListOfEnum "${pname}: theme variants" availableThemeVariants themeVariants
lib.checkListOfEnum "${pname}: catppuccin color variants" availableColorVariants.Catppuccin catppuccinColorVariants
lib.checkListOfEnum "${pname}: dracula color variants" availableColorVariants.Dracula draculaColorVariants
lib.checkListOfEnum "${pname}: gruvbox color variants" availableColorVariants.Gruvbox  gruvboxColorVariants
lib.checkListOfEnum "${pname}: original color variants" availableColorVariants.Original originalColorVariants

stdenvNoCC.mkDerivation {
  inherit pname;
  version = "unstable-2023-10-04";

  src = fetchFromGitHub {
    owner = "TeddyBearKilla";
    repo = "Afterglow-Cursors-Recolored";
    rev = "940a5d30e52f8c827fa249d2bbcc4af889534888";
    hash = "sha256-GR+d+jrbeIGpqal5krx83PxuQto2PQTO3unQ+jaJf6s=";
  };

  installPhase = let
    dist = {
      Catppuccin = "cat";
      Dracula = "dracula";
      Gruvbox = "gruvbox";
    };
    withAlternate = xs: xs': if xs != [ ] then xs else xs';
    themeVariants' = withAlternate themeVariants availableThemeVariants;
    colorVariants = {
      Catppuccin =  withAlternate catppuccinColorVariants availableColorVariants.Catppuccin;
      Dracula = withAlternate draculaColorVariants availableColorVariants.Dracula;
      Gruvbox = withAlternate gruvboxColorVariants availableColorVariants.Gruvbox;
      Original = withAlternate originalColorVariants availableColorVariants.Original;
    };
  in ''
    runHook preInstall

    ${
      lib.concatMapStringsSep "\n" (theme:
        lib.concatMapStringsSep "\n" (color: ''
          mkdir -p $out/share/icons/Afterglow-Recolored-${theme}-${color}
          cp -rT colors/${theme}/${color}/dist-${lib.optionalString (theme != "Original") (dist.${theme} + "-")}${lib.toLower color} $out/share/icons/Afterglow-Recolored-${theme}-${color}
        '') colorVariants.${theme}
      ) themeVariants'
    }

    runHook postInstall
  '';

  meta = with lib; {
    description = "Recolors of the original Afterglow Cursors";
    homepage = "https://github.com/TeddyBearKilla/Afterglow-Cursors-Recolored";
    maintainers = with maintainers; [ d3vil0p3r ];
    platforms = platforms.all;
    license = licenses.gpl3Plus;
  };
}
```

Reply:
I would keep that level of granularity because I'm thinking about a use case where a user has its theme, for example a Green Theme, and it will use only, i.e., Dracula Green cursor color. The other ones will remain in the system just to occupy disk space as unused resources.

Reply:
I'm not entirely sure that would be realistically be that much less storage used though. For the user to even build the package, they would first have to download the entire source of the package into the nix store.

Reply:
As you wish. If you would like to remove the implemented granularity, I can edit it.

Reply:
Hmm. Let's maybe leave it for now, and see if there are any more comments about it.

Reply:
@D3vil0p3r @h7x4 In my opinion, even if it may not save a significant amount of disk space, it's still beneficial to provide those options. Certainly, the system will initially fetch everything, but the user might not want to include specific color themes within their system. Conversely, if the user desires to have all included, they have the freedom to choose that as well.

Reply:
Okay, how about this - we symlink the directories from `$src` instead?

Reply:
> Okay, how about this - we symlink the directories from `$src` instead?

Do you mean by adding `$src` as `cp -rT $src/colors/${theme}/${color}/dist-${lib.optionalString (theme != "Original") (dist.${theme} + "-")}${lib.toLower color} $out/share/icons/Afterglow-Recolored-${theme}-${color}`?

Reply:
Almost, but rather with `ln -s`. (Added newlines for a bit more readabilty)

```nix
    ${
      lib.concatMapStringsSep "\n" (theme:
        lib.concatMapStringsSep "\n" (color: ''
          ln -s \
            "$src/colors/${theme}/${color}/dist-${lib.optionalString (theme != "Original") (dist.${theme} + "-")}${lib.toLower color}" \
            "$out/share/icons/Afterglow-Recolored-${theme}-${color}"
        '') colorVariants.${theme}
      ) themeVariants'
    }
```

Reply:
What is the advantage to use `ln -s` instead of `cp -T`?

Reply:
I'm having a bit of a hard time understanding the `-T` flag from the manpage, but `ln -s` will make a symlink (minimal pointer or shortcut that just references the real location of the directory or file), such that we don't keep two copies of the same files.

Reply:
> I'm having a bit of a hard time understanding the `-T` flag from the manpage, but `ln -s` will make a symlink (minimal pointer or shortcut that just references the real location of the directory or file), such that we don't keep two copies of the same files.

What I don't understand is the resulting directories. I expect to have `cursors` and `index.theme` inside:
```
result/share/icons/Afterglow-Recolored-Catppuccin-Flamingo
```
instead, they are in:
```
result/share/icons/Afterglow-Recolored-Catppuccin-Flamingo/dist-cat-flamingo
```

Reply:
Even with the `mkdir` removed?

Reply:
> Even with the `mkdir` removed?

Without `mkdir` it produces error.

Reply:
I think it should be fine if you run `mkdir -p "$out/share/icons"` before creating the symlinks.

Reply:
> I think it should be fine if you run `mkdir -p "$out/share/icons"` before creating the symlinks.

It works. I pushed the changes.